pkg/app/piped/trigger/deployment.go

```diff
@@ -183,6 +185,21 @@ func buildDeployment(
 	return deployment, nil
 }
 
+func (t *Trigger) reportDeploymentFailed(reason string) {
+	defer func() {
+		t.notifier.Notify(model.NotificationEvent{
+			Type: model.NotificationEventType_EVENT_PIPED_FAILED,
```

Comment:
I don't think that event is related to `Piped` or belong to the `PIPED` event group.
Instead, it should be a kind of `DEPLOYMENT` group. More specifically it would be `NotificationEventType_EVENT_DEPLOYMENT_TRIGGER_FAILED`.

Reply:
One vote for `NotificationEventType_EVENT_DEPLOYMENT_TRIGGER_FAILED`

And I'm not sure why this is enclosed by the `defer` block.

Reply:
> And I'm not sure why this is enclosed by the defer block.

Oh, that's my mistake. I'll fix it, too.